code/controllers/subsystems/job.dm

```diff
@@ -425,32 +425,49 @@
 /mob/living/carbon/human
 	var/tmp/odin_despawn_timer
 
-/mob/living/carbon/human/proc/odin_timeout()
+/mob/living/proc/odin_timeout()
 	if (!istype(get_area(src), /area/centcom/spawning))
 		return
 
 	if (!client)
 		SSjobs.odin_despawn_mob(src)
-	else
+	else if(ishuman(src))
```

Comment:
Hnrg.

`istype(src)` is bad and usually demonstrates bad objective oriented programming. Ideally, you wouldn't write code with those checks present. This is how to refactor it:
```
/mob/living/proc/odin_timeout()
	if (!istype(get_area(src), /area/centcom/spawning))
		return FALSE

	if (!client)
		SSjobs.odin_despawn_mob(src)
		return FALSE

	return TRUE

/mob/living/carbon/human/odin_timeout()
	. = ..()

	if (!.)
		return FALSE

	var/datum/spawnpoint/spawnpos = spawntypes["Cryogenic Storage"]
	if(spawnpos && istype(spawnpos))
		src << "<span class='warning'>You come to the sudden realization that you never left the Aurora at all! You were in cryo the whole time!</span>"
		src.forceMove(pick(spawnpos.turfs))
		global_announcer.autosay("[real_name], [mind.role_alt_title], [spawnpos.msg].", "Cryogenic Oversight")
		if(!src.megavend)
			var/rank = src.mind.assigned_role
			SSjobs.EquipRank(src, rank, 1)
			src.megavend = TRUE
	else
		SSjobs.odin_despawn_mob(src) //somehow they can't spawn at cryo, so this is the only recourse of action.

	return TRUE

/mob/living/silicon/robot/odin_timeout()
	. = ..()

	if (!.)
		return FALSE

	var/datum/spawnpoint/spawnpos = spawntypes["Cyborg Storage"]
	if(spawnpos && istype(spawnpos))
		src << "<span class='warning'>You come to the sudden realization that you never left the Aurora at all! You were in robotic storage the whole time!</span>"
		src.forceMove(pick(spawnpos.turfs))
		global_announcer.autosay("[real_name], [mind.role_alt_title], [spawnpos.msg].", "Robotic Oversight")
	else
		SSjobs.odin_despawn_mob(src)
```

The reason for this is cleanliness and extendability. It makes special cases much more logical to handle, and fall in line with how OOP is usually handled anyways.